### PR TITLE
Add p-projection for mortars for global time-stepping

### PR DESCRIPTION
**Pull request by kidder**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp [hunk 1]
```diff
@@ -150,4 +152,124 @@ struct Mortars {
...
+          normal_covector_and_magnitude,
+      const gsl::not_null<mortar_data_history_type*>
+      /*mortar_data_history*/,
+      const Mesh<dim>& /*new_mesh*/, const Element<dim>& new_element,
+      const std::pair<Mesh<dim>, Element<dim>>& /*old_mesh_and_element*/) {
+    static_assert(not Metavariables::local_time_stepping,
+                  "AMR with local time-stepping is not yet supported");
+
+    mortar_data->clear();
+    mortar_mesh->clear();
```
Comment (nilsvu) [inline src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp:210]: How is this set later? My understanding is that we can set this here by keeping track of the neighbor meshes, correct?

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp (hunk 1 above)
Comment (kidder) [reply src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp:210]: It is set when the mortar data is received from the neighbor.  Needs to be reactive in order for subcell to work.

Comment (kidder) [reply src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp:210]: Do you have a specific change request?

File: src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp [hunk 2]
```diff
@@ -150,4 +152,124 @@ struct Mortars {
...
+  using dt_variables_tag = typename db::add_tag_prefix<
+      ::Tags::dt, typename Metavariables::system::variables_tag>;
+  using mortar_data_history_type =
+      typename Tags::MortarDataHistory<dim,
+                                       typename dt_variables_tag::type>::type;
+
+  using Key = std::pair<Direction<dim>, ElementId<dim>>;
+
+  template <typename MappedType>
+  using MortarMap = std::unordered_map<Key, MappedType, boost::hash<Key>>;
```
Comment (nilsvu) [inline src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp]: (optional) These are `::dg::MortarId<Dim>` and `::dg::MortarMap<Dim, MappedType`, want to use those directly? (in MortarHelpers.hpp, which you already include)

**Review by nilsvu — DISMISSED** [review]

File: src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp (hunk 1 above)
Comment (nilsvu) [reply src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp:210]: Nope, sounds good.

File: src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp (hunk 2 above)
Comment (kidder) [reply src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp]: okay, rebased and squashed this optional change

**Review by nilsvu — APPROVED** [review]

**Pull request merged**